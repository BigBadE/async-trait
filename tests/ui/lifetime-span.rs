use async_trait::async_trait;

struct A;
struct B;

#[async_trait]
pub trait Trait<'r> {
    async fn method(&'r self);
}

#[async_trait]
impl Trait for A {
    async fn method(&self) { }
}

#[async_trait]
impl<'r> Trait<'r> for B {
    async fn method(&self) { }
}

fn main() {}
