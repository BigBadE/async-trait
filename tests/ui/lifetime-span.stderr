error[E0726]: implicit elided lifetime not allowed here
  --> $DIR/lifetime-span.rs:12:6
   |
12 | impl Trait for A {
   |      ^^^^^- help: indicate the anonymous lifetime: `<'_>`

error[E0195]: lifetime parameters or bounds on method `method` do not match the trait declaration
  --> $DIR/lifetime-span.rs:11:1
   |
6  | #[async_trait]
   | -------------- lifetimes in impl do not match this method in trait
...
11 | #[async_trait]
   | ^^^^^^^^^^^^^^ lifetimes do not match method in trait
   |
   = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0195]: lifetime parameters or bounds on method `method` do not match the trait declaration
  --> $DIR/lifetime-span.rs:16:1
   |
6  | #[async_trait]
   | -------------- lifetimes in impl do not match this method in trait
...
16 | #[async_trait]
   | ^^^^^^^^^^^^^^ lifetimes do not match method in trait
   |
   = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)
